### DISCUSSION TITLE
Strange Macro behaviour

### DISCUSSION THREAD

**Discussion by eliascotto** [borkdude/sci#607] [Q&A] [answered]
Hey, I've using SCI and it work perfectly for simply evaluating strings in the browser. Thanks for the project!

I just need to use this particular macro in the browser. I'm using `0.2.6` with the latest Clojurescript. This macro simply declare a local var with the value get from the call of `get-context` and use it as a first parameter when calling all the functions in the `forms`.

```clojure
(def with-context ^:sci/macro
  (fn [_&form _&env & forms]
    (let [gx (gensym)]
      `(let [~gx (get-context)]
         ~@(map (fn [f]
                  (if (seq? f)
                    `(~(first f) ~gx ~@(next f))
                    `(~f ~gx)))
                forms)
         ~gx))))
```

I tried the macro in a Clojure REPL and works correctly
```
user=> (defn get-context [] 3)
user=> (macroexpand '(with-context (+ 2) (+ 3)))
(let* [G__179 (user/get-context)] (+ G__179 2) (+ G__179 3) G__179)
```
while using it in SCI seems to don't pass the value correctly.

**Comment by borkdude** [comment, marked as answer]
@elias94 I think the problem is that you're defining the macro in some CLJS namespace `your-ns` and `(get-context)` gets expanded into `(your-ns/get-context)`. If `your-ns` doesn't match the namespace you are defining `get-context` in in the SCI ctx, then this doesn't match up. Perhaps this is sufficient information. If not, please provide a full repro with a failing example and an expected outcome.
